publication/src/test/scala/hmda/publication/reports/disclosure/D51Spec.scala

```diff
@@ -0,0 +1,65 @@
+package hmda.publication.reports.disclosure
+
+import akka.NotUsed
+import akka.actor.ActorSystem
+import akka.stream.ActorMaterializer
+import akka.stream.scaladsl.Source
+import hmda.model.fi.lar.{ LarGenerators, LoanApplicationRegister }
+import hmda.model.publication.reports.ActionTakenTypeEnum._
+import hmda.model.publication.reports.ApplicantIncomeEnum.LessThan50PercentOfMSAMedian
+import hmda.model.publication.reports.{ EthnicityBorrowerCharacteristic, MSAReport, MinorityStatusBorrowerCharacteristic, RaceBorrowerCharacteristic }
+import hmda.query.model.filing.LoanApplicationRegisterQuery
+import hmda.query.repository.filing.LarConverter._
+import org.scalacheck.Gen
+import org.scalatest.{ MustMatchers, WordSpec }
+
+import scala.concurrent.Await
+import scala.concurrent.duration._
+
+class D51Spec extends WordSpec with MustMatchers with LarGenerators {
+
+  implicit val system = ActorSystem()
+  implicit val ec = system.dispatcher
+  implicit val materializer = ActorMaterializer()
+
+  val respId = "98765"
+  val fips = 18700 //Corvallis, OR
+  def loanType = Gen.oneOf(2, 3, 4).sample.get
+  def propType = Gen.oneOf(1, 2).sample.get
+
+  val lars = lar100ListGen.sample.get.map { lar: LoanApplicationRegister =>
+    val geo = lar.geography.copy(msa = fips.toString)
+    val loan = lar.loan.copy(loanType = loanType, propertyType = propType, purpose = 1)
+    lar.copy(respondentId = respId, geography = geo, loan = loan)
+  }
+
+  val source: Source[LoanApplicationRegisterQuery, NotUsed] = Source
+    .fromIterator(() => lars.toIterator)
+    .map(lar => toLoanApplicationRegisterQuery(lar))
+
+  val expectedDispositions = List(ApplicationReceived, LoansOriginated, ApprovedButNotAccepted, ApplicationsDenied, ApplicationsWithdrawn, ClosedForIncompleteness)
+
+  "Generate a Disclosure 5-1 report" in {
+    val result = Await.result(D51.generate(source, fips, respId), 5.seconds)
+
+    result.msa mustBe MSAReport("18700", "Corvallis, OR", "OR", "Oregon")
+    result.table mustBe "5-1"
+    result.respondentId mustBe "98765"
+    result.applicantIncomes.size mustBe 5
+
+    val lowestIncome = result.applicantIncomes.head
+    lowestIncome.applicantIncome mustBe LessThan50PercentOfMSAMedian
```

Comment:
This test is just verifying that the result has the correct structure. Even if there aren't any LARs that fall into that income category, there will still be a row of data for the `LessThan50PercentOfMSAMedian` category (it'll just have zeros as its data).